content/blog/case-weights/index.Rmd

```diff
@@ -0,0 +1,310 @@
+
+This is also an area where we'd like [community feedback](https://community.rstudio.com/). 
+
+## Tidymodels syntax
+
+Let's work through an example. We'll use some data simulated with a severe class imbalance. These functions are in the new version of the modeldata package
+
+```{r sim-data}
+set.seed(1)
+training_sim <- sim_classification(5000, intercept = -25) 
+training_sim %>% count(class)
+```
+
+If we would like to encourage models to more accurately predict the minority class, we can give these samples a much larger weight in the analysis
+
+```{r sim-wts}
+training_sim <-
+  training_sim %>% 
+  mutate(
+    case_wts = ifelse(class == "class_1", 60, 1),
+    case_wts = importance_weights(case_wts)
+  )
+```
+
+We strongly advise that users set the case weight column before any other tidymodels functions are used. This ensures that they are handled correctly in the analyses that follow. 
+
+Let's use 10-fold cross-validation to resample the data. This case is unaffected by the presence of weights: 
+
+```{r sim-folds}
+set.seed(2)
+sim_folds <- vfold_cv(training_sim, strata = class)
+```
+
+We'll fit a regularized logistic regression model to the data using glmnet: 
+
+```{r sim-model}
+lr_spec <- 
+  logistic_reg(penalty = tune(), mixture = 1) %>% 
+  set_engine("glmnet")
+```
+
+For this model, we need to ensure that the predictors are in the same units. We'll use a recipe to center and scale the data and also add some spline terms for predictors that appear to have a nonlinear relationship with the outcome: 
+
+```{r sim-rec}
+sim_rec <- 
+  recipe(class ~ ., data = training_sim) %>% 
+  step_ns(starts_with("non_linear"), deg_free = 10) %>% 
+  step_normalize(all_numeric_predictors())
+sim_rec
+```
+
+There are a few things to point out here. The recipe automatically detects the case weights even though they are captured by the dot in the right-hand side of the formula. The recipe automatically sets their role and will error if that column is changed in any way. 
+
+As mentioned above, any unsupervised steps are unaffected by importance weights so neither `step_ns()` or `step_normalize()` use the weights in their calculations. 
+
+When using case weights, we would like to encourage users to keep their model and preprocessing tool within a workflow. The workflows package now has a `add_case_weights()` function to help here: 
+
+```{r sim-wflow}
+lr_wflow <- 
+  workflow() %>% 
+  add_model(lr_spec) %>% 
+  add_recipe(sim_rec) %>% 
+  add_case_weights(case_wts)
+lr_wflow
+```
+
+From here, let's tune the model. This code doesn't have to change at all. We will add some additional metrics to show the affect of the weights though: 
```

Comment:
```suggestion
From here, let's tune the model. This code doesn't have to change at all. Because we set the case weights column with `add_case_weights()`, tune and workflows automatically know how to pass the case weights on to the underlying parsnip model. We will add some additional metrics to show the affect of the weights though: 
```